packages/flutter/lib/src/widgets/view.dart

```diff
@@ -188,24 +188,43 @@ class _ViewState extends State<View> with WidgetsBindingObserver {
     debugLabel: kReleaseMode ? null : 'View Scope',
   );
   final FocusTraversalPolicy _policy = ReadingOrderTraversalPolicy();
+  bool _viewHasFocus = false;
 
   @override
   void initState() {
     super.initState();
     WidgetsBinding.instance.addObserver(this);
+    _scopeNode.addListener(_scopeFocusChangeListener);
   }
 
   @override
   void dispose() {
     WidgetsBinding.instance.removeObserver(this);
+    _scopeNode.removeListener(_scopeFocusChangeListener);
     _scopeNode.dispose();
     super.dispose();
   }
 
+  void _scopeFocusChangeListener() {
+    if (_viewHasFocus == _scopeNode.hasFocus) {
+      return;
+    }
+    // Scope focus changed, and it doesn't match the view focus, so inform the
+    // view so it knows to change its focus.
+    WidgetsBinding.instance.platformDispatcher.requestViewFocusChange(
```

Comment:
If you have 2 views, and focus transitions from view A -> view B
I'd expect to see only one call 

```
requestViewFocusChange(
  viewId: viewB.viewId,
  state: ViewFocusState.focused,
);
```

but instead, with this approach we'll get two calls (am i right?):
```
requestViewFocusChange(
  viewId: viewA.viewId,
  state: ViewFocusState.unfocused,
);

requestViewFocusChange(
  viewId: viewA.viewId,
  state: ViewFocusState.focused,
);
```

The issue with this approach is that we'll be "unnecesary" blurring view `A` before focusing view `B`. Not the end of the world. Is it guaranteed that things will run in this order btw? If the second one happens first, it will be even more problematic.

Reply:
Did you mean this?

```dart
requestViewFocusChange(
  viewId: viewA.viewId,
  state: ViewFocusState.unfocused,
);

requestViewFocusChange(
  viewId: viewB.viewId,
  state: ViewFocusState.focused,
);
```

If so, then yes. I would suggest we just request only on `focused`: there can be only one focused view anyhow, and would avoid all of those issues.

Reply:
Okay, I modified it to only notify if the view should gain focus.